google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1alpha2/StreamWriter.java

```
@@ -0,0 +1,856 @@
+   *     }
+   *   }
+   *
+   *   public void onFailure(Throwable t) {
+   *     System.out.println("failed to write: " + t);
+   *   }
+   * }, MoreExecutors.directExecutor());
+   * }</pre>
+   *
+   * @param message the message in serialized format to write to BigQuery.
+   * @return the message ID wrapped in a future.
+   */
+  public ApiFuture<AppendRowsResponse> append(AppendRowsRequest message) {
+    Preconditions.checkState(!shutdown.get(), "Cannot append on a shut-down writer.");
+
+    final AppendRequestAndFutureResponse outstandingAppend =
+        new AppendRequestAndFutureResponse(message);
+    List<InflightBatch> batchesToSend;
+    messagesBatchLock.lock();
+    try {
+      batchesToSend = messagesBatch.add(outstandingAppend);
+      // Setup the next duration based delivery alarm if there are messages batched.
+      setupAlarm();
+      if (!batchesToSend.isEmpty()) {
+        for (final InflightBatch batch : batchesToSend) {
+          LOG.fine("Scheduling a batch for immediate sending.");
+          writeBatch(batch);
+        }
+      }
+    } finally {
+      messagesBatchLock.unlock();
+    }
+
+    return outstandingAppend.appendResult;
+  }
+
+  public void refreshAppend() throws IOException {
+    synchronized (this) {
+      Preconditions.checkState(!shutdown.get(), "Cannot append on a shut-down writer.");
+      if (stub != null) {
+        stub.shutdown();
+      }
+      backgroundResourceList.remove(stub);
+      stub = BigQueryWriteClient.create(stubSettings);
+      backgroundResourceList.add(stub);
+      backgroundResources = new BackgroundResourceAggregation(backgroundResourceList);
+      messagesBatch.resetAttachSchema();
+      bidiStreamingCallable = stub.appendRowsCallable();
+      clientStream = bidiStreamingCallable.splitCall(responseObserver);
+      try {
+        while (!clientStream.isSendReady()) {
+          Thread.sleep(100);
+        }
+      } catch (InterruptedException e) {
```

is there a reason why we're ignoring this exception? see: https://google.github.io/styleguide/javaguide.html#s6.2-caught-exceptions

---

This is to wait for channel to be ready. It is a common practice to wait on such things. If there are better guidelines on how to use the streaming API, please let me know.

---

I guess I was asking why we're catching an exception but not doing anything with it? see https://google.github.io/styleguide/javaguide.html#s6.2-caught-exceptions for best practice on exception handling -- if it's expected, maybe use `InterruptedException expected` instead of just `InterruptedException e` to be more descriptive?

---

Done.